composer/trainer/trainer_hparams.py

```diff
@@ -343,35 +343,35 @@ class TrainerHparams(hp.Hparams):
 
     # Save Checkpoint
     save_folder: Optional[str] = hp.auto(Trainer, 'save_folder')
-    save_filename: str = hp.auto(Trainer, 'save_filename')
-    save_artifact_name: str = hp.auto(Trainer, 'save_artifact_name')
-    save_latest_filename: str = hp.auto(Trainer, 'save_latest_filename')
-    save_latest_artifact_name: str = hp.auto(Trainer, 'save_latest_artifact_name')
-    save_overwrite: bool = hp.auto(Trainer, 'save_overwrite')
-    save_weights_only: bool = hp.auto(Trainer, 'save_weights_only')
-    save_interval: str = hp.auto(Trainer, 'save_interval')
-    save_num_checkpoints_to_keep: int = hp.auto(Trainer, 'save_num_checkpoints_to_keep')
+    save_filename: Optional[str] = hp.auto(Trainer, 'save_filename')
+    save_artifact_name: Optional[str] = hp.auto(Trainer, 'save_artifact_name')
```

Comment:
This change is fine, but the same change needs to be made in the Trainer signature. The trainer docstring will also need to be updated to reflect that setting this parameter to None means that checkpoints will not be logged as artifacts.